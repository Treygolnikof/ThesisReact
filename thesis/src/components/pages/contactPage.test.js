import React from 'react';
import ContactPage from './contactPage';
import {shallow} from 'enzyme';

describe('Testing <ContactPage/>', () => {
    const item = shallow(<ContactPage/>);

    it('ContactPage have rendered correctly', () => {
        expect(item).toMatchSnapshot();
    })

    it('ContactPage state "error" is false', () => {
        expect(item.state().error).toBeFalsy();
    })

    it('ContactPage state "loading" is true', () => {
        expect(item.state().loading).toBeFalsy();
    })

    it('ContactPage state "success" is true', () => {
        expect(item.state().success).toBeFalsy();
    })

    it('ContactPage state "item" is empty object', () => {
        expect(item.state().value).toBeNil();
    })
    
    it('Testing onSuccess', () => {
        item.instance().onSuccess();
        expect(item.state().success).toBeTruthy();
        expect(item.state().loading).toBeFalsy();
    })

    it('Testing onError', () => {
        item.instance().onError();
        expect(item.state().error).toBeTruthy();
        expect(item.state().loading).toBeFalsy();
    })

    it('Testing onBack', () => {
        item.instance().onBack();
        expect(item.state().success).toBeFalsy();
    })
})