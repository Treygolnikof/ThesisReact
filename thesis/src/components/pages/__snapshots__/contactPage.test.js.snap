// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <ContactPage/> ContactPage have rendered correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ContactPage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <div
          className="contact__banner"
        >
          <Container
            tag="div"
          >
            <Row
              tag="div"
            >
              <Col
                lg="6"
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <NavPanel
                  color="text-white"
                  img="Logo.svg"
                  position="header"
                />
              </Col>
            </Row>
            <h1
              className="title-big"
            >
              Contact us
            </h1>
          </Container>
        </div>,
        <section
          className="shop"
        >
          <Container
            tag="div"
          >
            <Row
              tag="div"
            >
              <Col
                lg={
                  Object {
                    "offset": 2,
                    "size": 8,
                  }
                }
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <div
                  className="title"
                >
                  Tell us about your taste
                </div>
                <img
                  alt="Beans logo"
                  className="beanslogo"
                  src="Beans_logo_dark.svg"
                />
                <GetForm
                  handleSubmit={[Function]}
                  phoneNumberMask={[Function]}
                  state={null}
                />
              </Col>
            </Row>
          </Container>
        </section>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Container
            tag="div"
          >
            <Row
              tag="div"
            >
              <Col
                lg="6"
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <NavPanel
                  color="text-white"
                  img="Logo.svg"
                  position="header"
                />
              </Col>
            </Row>
            <h1
              className="title-big"
            >
              Contact us
            </h1>
          </Container>,
          "className": "contact__banner",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Row
                tag="div"
              >
                <Col
                  lg="6"
                  tag="div"
                  widths={
                    Array [
                      "xs",
                      "sm",
                      "md",
                      "lg",
                      "xl",
                    ]
                  }
                >
                  <NavPanel
                    color="text-white"
                    img="Logo.svg"
                    position="header"
                  />
                </Col>
              </Row>,
              <h1
                className="title-big"
              >
                Contact us
              </h1>,
            ],
            "tag": "div",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <Col
                  lg="6"
                  tag="div"
                  widths={
                    Array [
                      "xs",
                      "sm",
                      "md",
                      "lg",
                      "xl",
                    ]
                  }
                >
                  <NavPanel
                    color="text-white"
                    img="Logo.svg"
                    position="header"
                  />
                </Col>,
                "tag": "div",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <NavPanel
                    color="text-white"
                    img="Logo.svg"
                    position="header"
                  />,
                  "lg": "6",
                  "tag": "div",
                  "widths": Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ],
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "color": "text-white",
                    "img": "Logo.svg",
                    "position": "header",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Contact us",
                "className": "title-big",
              },
              "ref": null,
              "rendered": "Contact us",
              "type": "h1",
            },
          ],
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Container
            tag="div"
          >
            <Row
              tag="div"
            >
              <Col
                lg={
                  Object {
                    "offset": 2,
                    "size": 8,
                  }
                }
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <div
                  className="title"
                >
                  Tell us about your taste
                </div>
                <img
                  alt="Beans logo"
                  className="beanslogo"
                  src="Beans_logo_dark.svg"
                />
                <GetForm
                  handleSubmit={[Function]}
                  phoneNumberMask={[Function]}
                  state={null}
                />
              </Col>
            </Row>
          </Container>,
          "className": "shop",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Row
              tag="div"
            >
              <Col
                lg={
                  Object {
                    "offset": 2,
                    "size": 8,
                  }
                }
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <div
                  className="title"
                >
                  Tell us about your taste
                </div>
                <img
                  alt="Beans logo"
                  className="beanslogo"
                  src="Beans_logo_dark.svg"
                />
                <GetForm
                  handleSubmit={[Function]}
                  phoneNumberMask={[Function]}
                  state={null}
                />
              </Col>
            </Row>,
            "tag": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Col
                lg={
                  Object {
                    "offset": 2,
                    "size": 8,
                  }
                }
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <div
                  className="title"
                >
                  Tell us about your taste
                </div>
                <img
                  alt="Beans logo"
                  className="beanslogo"
                  src="Beans_logo_dark.svg"
                />
                <GetForm
                  handleSubmit={[Function]}
                  phoneNumberMask={[Function]}
                  state={null}
                />
              </Col>,
              "tag": "div",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <div
                    className="title"
                  >
                    Tell us about your taste
                  </div>,
                  <img
                    alt="Beans logo"
                    className="beanslogo"
                    src="Beans_logo_dark.svg"
                  />,
                  null,
                  null,
                  <GetForm
                    handleSubmit={[Function]}
                    phoneNumberMask={[Function]}
                    state={null}
                  />,
                  null,
                ],
                "lg": Object {
                  "offset": 2,
                  "size": 8,
                },
                "tag": "div",
                "widths": Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Tell us about your taste",
                    "className": "title",
                  },
                  "ref": null,
                  "rendered": "Tell us about your taste",
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "Beans logo",
                    "className": "beanslogo",
                    "src": "Beans_logo_dark.svg",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                null,
                null,
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "handleSubmit": [Function],
                    "phoneNumberMask": [Function],
                    "state": null,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                null,
              ],
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
        "type": "section",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div
            className="contact__banner"
          >
            <Container
              tag="div"
            >
              <Row
                tag="div"
              >
                <Col
                  lg="6"
                  tag="div"
                  widths={
                    Array [
                      "xs",
                      "sm",
                      "md",
                      "lg",
                      "xl",
                    ]
                  }
                >
                  <NavPanel
                    color="text-white"
                    img="Logo.svg"
                    position="header"
                  />
                </Col>
              </Row>
              <h1
                className="title-big"
              >
                Contact us
              </h1>
            </Container>
          </div>,
          <section
            className="shop"
          >
            <Container
              tag="div"
            >
              <Row
                tag="div"
              >
                <Col
                  lg={
                    Object {
                      "offset": 2,
                      "size": 8,
                    }
                  }
                  tag="div"
                  widths={
                    Array [
                      "xs",
                      "sm",
                      "md",
                      "lg",
                      "xl",
                    ]
                  }
                >
                  <div
                    className="title"
                  >
                    Tell us about your taste
                  </div>
                  <img
                    alt="Beans logo"
                    className="beanslogo"
                    src="Beans_logo_dark.svg"
                  />
                  <GetForm
                    handleSubmit={[Function]}
                    phoneNumberMask={[Function]}
                    state={null}
                  />
                </Col>
              </Row>
            </Container>
          </section>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Container
              tag="div"
            >
              <Row
                tag="div"
              >
                <Col
                  lg="6"
                  tag="div"
                  widths={
                    Array [
                      "xs",
                      "sm",
                      "md",
                      "lg",
                      "xl",
                    ]
                  }
                >
                  <NavPanel
                    color="text-white"
                    img="Logo.svg"
                    position="header"
                  />
                </Col>
              </Row>
              <h1
                className="title-big"
              >
                Contact us
              </h1>
            </Container>,
            "className": "contact__banner",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <Row
                  tag="div"
                >
                  <Col
                    lg="6"
                    tag="div"
                    widths={
                      Array [
                        "xs",
                        "sm",
                        "md",
                        "lg",
                        "xl",
                      ]
                    }
                  >
                    <NavPanel
                      color="text-white"
                      img="Logo.svg"
                      position="header"
                    />
                  </Col>
                </Row>,
                <h1
                  className="title-big"
                >
                  Contact us
                </h1>,
              ],
              "tag": "div",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <Col
                    lg="6"
                    tag="div"
                    widths={
                      Array [
                        "xs",
                        "sm",
                        "md",
                        "lg",
                        "xl",
                      ]
                    }
                  >
                    <NavPanel
                      color="text-white"
                      img="Logo.svg"
                      position="header"
                    />
                  </Col>,
                  "tag": "div",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": <NavPanel
                      color="text-white"
                      img="Logo.svg"
                      position="header"
                    />,
                    "lg": "6",
                    "tag": "div",
                    "widths": Array [
                      "xs",
                      "sm",
                      "md",
                      "lg",
                      "xl",
                    ],
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "color": "text-white",
                      "img": "Logo.svg",
                      "position": "header",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Contact us",
                  "className": "title-big",
                },
                "ref": null,
                "rendered": "Contact us",
                "type": "h1",
              },
            ],
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Container
              tag="div"
            >
              <Row
                tag="div"
              >
                <Col
                  lg={
                    Object {
                      "offset": 2,
                      "size": 8,
                    }
                  }
                  tag="div"
                  widths={
                    Array [
                      "xs",
                      "sm",
                      "md",
                      "lg",
                      "xl",
                    ]
                  }
                >
                  <div
                    className="title"
                  >
                    Tell us about your taste
                  </div>
                  <img
                    alt="Beans logo"
                    className="beanslogo"
                    src="Beans_logo_dark.svg"
                  />
                  <GetForm
                    handleSubmit={[Function]}
                    phoneNumberMask={[Function]}
                    state={null}
                  />
                </Col>
              </Row>
            </Container>,
            "className": "shop",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Row
                tag="div"
              >
                <Col
                  lg={
                    Object {
                      "offset": 2,
                      "size": 8,
                    }
                  }
                  tag="div"
                  widths={
                    Array [
                      "xs",
                      "sm",
                      "md",
                      "lg",
                      "xl",
                    ]
                  }
                >
                  <div
                    className="title"
                  >
                    Tell us about your taste
                  </div>
                  <img
                    alt="Beans logo"
                    className="beanslogo"
                    src="Beans_logo_dark.svg"
                  />
                  <GetForm
                    handleSubmit={[Function]}
                    phoneNumberMask={[Function]}
                    state={null}
                  />
                </Col>
              </Row>,
              "tag": "div",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <Col
                  lg={
                    Object {
                      "offset": 2,
                      "size": 8,
                    }
                  }
                  tag="div"
                  widths={
                    Array [
                      "xs",
                      "sm",
                      "md",
                      "lg",
                      "xl",
                    ]
                  }
                >
                  <div
                    className="title"
                  >
                    Tell us about your taste
                  </div>
                  <img
                    alt="Beans logo"
                    className="beanslogo"
                    src="Beans_logo_dark.svg"
                  />
                  <GetForm
                    handleSubmit={[Function]}
                    phoneNumberMask={[Function]}
                    state={null}
                  />
                </Col>,
                "tag": "div",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    <div
                      className="title"
                    >
                      Tell us about your taste
                    </div>,
                    <img
                      alt="Beans logo"
                      className="beanslogo"
                      src="Beans_logo_dark.svg"
                    />,
                    null,
                    null,
                    <GetForm
                      handleSubmit={[Function]}
                      phoneNumberMask={[Function]}
                      state={null}
                    />,
                    null,
                  ],
                  "lg": Object {
                    "offset": 2,
                    "size": 8,
                  },
                  "tag": "div",
                  "widths": Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Tell us about your taste",
                      "className": "title",
                    },
                    "ref": null,
                    "rendered": "Tell us about your taste",
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "alt": "Beans logo",
                      "className": "beanslogo",
                      "src": "Beans_logo_dark.svg",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "img",
                  },
                  null,
                  null,
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "handleSubmit": [Function],
                      "phoneNumberMask": [Function],
                      "state": null,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  null,
                ],
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
          "type": "section",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
